frc53_nft/src/state.rs

```diff
@@ -553,18 +585,134 @@ impl NFTState {
     }
 
     /// Get the metadata for a token
-    pub fn get_metadata<BS: Blockstore>(&self, bs: &BS, token_id: u64) -> Result<String> {
+    pub fn get_metadata<BS: Blockstore>(&self, bs: &BS, token_id: TokenID) -> Result<String> {
         let token_data_array = self.get_token_data_amt(bs)?;
         let token = token_data_array.get(token_id)?.ok_or(StateError::TokenNotFound(token_id))?;
         Ok(token.metadata.clone())
     }
 
     /// Get the owner of a token
-    pub fn get_owner<BS: Blockstore>(&self, bs: &BS, token_id: u64) -> Result<ActorID> {
+    pub fn get_owner<BS: Blockstore>(&self, bs: &BS, token_id: TokenID) -> Result<ActorID> {
         let token_data_array = self.get_token_data_amt(bs)?;
         let token = token_data_array.get(token_id)?.ok_or(StateError::TokenNotFound(token_id))?;
         Ok(token.owner)
     }
+
+    /// List all the minted tokens
+    pub fn list_tokens<BS: Blockstore>(
+        &self,
+        bs: &BS,
+        range_start: Option<AmtCursor>,
+        limit: Option<u64>,
+    ) -> Result<(TokenSet, Option<AmtCursor>)> {
+        let token_data_array = match &range_start {
+            Some(cursor) => self.get_token_amt_at_cursor(bs, cursor)?,
+            None => self.get_token_data_amt(bs)?,
```

Comment:
Nit: factor out this repeated pattern, perhaps by passing the optional cursor to get_toke_data_amt